mobile_verifier/src/banning/ingestor.rs

```diff
@@ -0,0 +1,165 @@
+            file_upload.clone(),
+            FileSinkCommitStrategy::Manual,
+            FileSinkRollTime::Default,
+            env!("CARGO_PKG_NAME"),
+        )
+        .await?;
+
+        let (report_rx, ingest_server) =
+            mobile_ban::report_source(pool.clone(), file_store.clone(), settings.start_after)
+                .await?;
+
+        let ingestor = Self {
+            pool,
+            auth_verifier,
+            report_rx,
+            verified_sink,
+        };
+
+        Ok(TaskManager::builder()
+            .add_task(verified_sink_server)
+            .add_task(ingest_server)
+            .add_task(ingestor)
+            .build())
+    }
+
+    pub fn new(
+        pool: PgPool,
+        auth_verifier: AuthorizationClient,
+        report_rx: BanReportSource,
+        verified_sink: VerifiedBanReportSink,
+    ) -> Self {
+        Self {
+            pool,
+            auth_verifier,
+            report_rx,
+            verified_sink,
+        }
+    }
+
+    async fn run(mut self, mut shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tracing::info!("starting ban ingestor");
+
+        loop {
+            tokio::select! {
+                biased;
+                _= &mut shutdown => break,
+                msg = self.report_rx.recv() => {
+                    let Some(file_info_stream) = msg else {
+                        anyhow::bail!("hotspot ban FileInfoPoller sender was dropped unexpectedly");
+                    };
+                    self.process_file(file_info_stream).await?;
+                }
+            }
+        }
+
+        tracing::info!("stopping ban ingestor");
+
+        Ok(())
+    }
+
+    async fn process_file(&self, file_info_stream: BanReportStream) -> anyhow::Result<()> {
+        let file = &file_info_stream.file_info.key;
+        tracing::info!(file, "processing");
+
+        let mut txn = self.pool.begin().await?;
+        let mut stream = file_info_stream.into_stream(&mut txn).await?;
+
+        while let Some(report) = stream.next().await {
+            let verified_report = process_ban_report(&mut txn, &self.auth_verifier, report).await?;
+            let status = verified_report.status.as_str_name();
+            self.verified_sink
+                .write(verified_report, &[("status", status)])
+                .await?;
+        }
+
+        self.verified_sink.commit().await?;
+        txn.commit().await?;
+
+        Ok(())
+    }
+}
+
+pub async fn process_ban_report(
+    conn: &mut PgConnection,
+    auth_verifier: &impl AuthorizationVerifier,
+    report: BanReport,
+) -> anyhow::Result<VerifiedBanReport> {
+    let status = get_verified_status(auth_verifier, &report.report.ban_key).await?;
+
+    let verified_report = VerifiedBanReport {
+        verified_timestamp: Utc::now(),
+        report,
+        status,
+    };
+
+    if verified_report.is_valid() {
+        db::update_hotspot_ban(conn, &verified_report).await?;
+    }
```

Comment:
The same question as above. Should we log (print) invalid reports?

Reply:
In this process we are creating a new report of `VerifiedBanIngestReportV1`, all Ban Requests exist in that report regardless of status.

We also have metrics being output when we write to the file sink.
```rs
self.verified_sink.write(verified_report, &[("status", status)]).await?;
```

I don't feel that outputting logs for individual statuses is useful at this point.
And tracking valid/invalid metrics for a given file would be adding code for information we could get from the output file itself, or from metrics we are already gathering.